### PR TITLE
Rewrite in python

### PR DESCRIPTION
**Pull request by MoralCode**
per discussion in #15 i went and YOLO rewrote this in python because I wanted to add IPv6 support and I was getting sick of the weird tcl syntax.

not yet tested. will also probably merge in more stuff to this branch too, like: an updated dockerfile and (at the maintainers option as to whether this belongs in this PR or a separate one) the ipv6 capabilities

### CODE REVIEW

File: nfsn-ddns.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+def output(msg, type_msg=None, timestamp=None):
+    if timestamp is None:
+        timestamp = datetime.now().strftime("%y-%m-%d %H:%M:%S")  
+    type_str = f"{type_msg}: " if type_msg is not None else ""
+    print(f"{timestamp}: {type_str}{msg}")
+    
+
+def validateNFSNResponse(response):
+    if response is None:
+        print("none response received")
```
Comment (mhum) [inline nfsn-ddns.py:39]: Do these need to be logged? These are all acceptable responses from the API. I _think_ at one point I had a verbose or debug log level in this project and these are the sorts of things I would include in a separate log level. But for normal runs, these probably aren't required if the update is successful.

File: nfsn-ddns.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
+from urllib.parse import urlencode
```
Comment (mhum) [inline nfsn-ddns.py:1]: It's unfortunate that [PEP8](https://peps.python.org/pep-0008/#imports) doesn't recommend how to order these, but it at least does recommend grouping the different types of imports. Once grouped, ordering alphabetically is the usual practice.

File: nfsn-ddns.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        output(response.get('debug'), type_msg="ERROR")
+
+
+def makeNFSNHTTPRequest(path, body, nfsn_username, nfsn_apikey):
+    url = NFSN_API_DOMAIN + path
+    headers = createNFSNAuthHeader(nfsn_username, nfsn_apikey, path, body)
+    headers["Content-Type"] = "application/x-www-form-urlencoded"
+
+    response = requests.post(url, data=body, headers=headers)
+    # response.raise_for_status()
```
Comment (mhum) [inline nfsn-ddns.py:63]: Leftover from trying some things during development?

**Review by mhum — APPROVED** [review]
Thanks so much for the re-write!  Only a couple of styling and superficial comments, so I'll accept this PR. I really appreciate keeping the code structure broken up into functions. Maybe I'll add tests at some point and this makes it so much easier. Also using types :heart_eyes: 

I've tested this running the script using `export` and with docker compose and the `environment` config (my setup). I'll assume the `.env` setup also works!

File: nfsn-ddns.py (hunk 1 above)
Comment (MoralCode) [reply nfsn-ddns.py:39]: oh i think this was just a leftover comment from my own debugging while trying to understand what possible values the NFSN response could be

File: nfsn-ddns.py (hunk 3 above)
Comment (MoralCode) [reply nfsn-ddns.py:63]: the raise for status comment? yeah pretty much

File: nfsn-ddns.py (hunk 2 above)
Comment (MoralCode) [reply nfsn-ddns.py:1]: yeah maybe adding a linter to the build step along with unit tests might be helpful to maintain this, cuz i sorta just throw imports up wherever and if it works it works, but i see how there may be value in following the standards somewhat

**Pull request merged**